Add option to calculate settling velocities for differently shaped particles

jormercury opened the pull request:
See:
vignettes/3.2-Particle-characteristics.md (https://github.com/rivm-syso/SBooScripts/blob/05c8b25a530d31693d472076cd2f3d882a565ee7/vignettes/3.2-Particle-characteristics.md)
Please when you have time @nadimsaadi add some more description on the approach used to calculate the Settling Velocity in the above vignette.
and
vignettes/Development/20250826_Plastics_Shape_update.md 
https://github.com/rivm-syso/SBooScripts/blob/05c8b25a530d31693d472076cd2f3d882a565ee7/vignettes/Development/20250826_Plastics_Shape_update.md

anne-hids committed:
Update k_Runoff.R

anne-hids committed:
Merge branch 'development' into Runoff_update_clean

Commit:
Introduce MutateHelper function

Commit:
Introduce Interception Fraction for size dependant particle Runoff adjustment

Commit:
Install SimpleBox

Commit:
Implementation of the Shape and Drag methods

Commit:
fix names

Commit:
debugging

jormercury committed:
First implementation tested for shapes

jormercury committed:
Update landscape combinations that do not exist in current scenario. return(NA)

jormercury committed:
Fix for some matrix density and particle variables

nadimsaadi commented on R/f_PerimeterParticle.R:
```diff
@@ -31,11 +38,11 @@ f_PerimeterParticle <- function(Shape, Longest_side = NULL, Intermediate_side =
     return(perimeter)
-  } else if (Shape == "Cylindric - circular") {
+  } else if (Shape == "Cylindric - circular" | Shape == "Fiber") {
     radius <- Longest_side / 2
```
Should this not be: radius = short_side / 2?

nadimsaadi replied:
this how it is defined in the f_SurfaceArea.

jormercury replied:
Yes, I think so. Good idea to go over them and check. A mistake with radius and diameter is easily made. And indeed good to check that Shortest_side and the other sides are used according to their definition in the Particle Characteristics documentation.

nadimsaadi commented on R/v_SettlingVelocity.R:
```diff
@@ -56,29 +65,52 @@ SettlingVelocity <- function(rad_species, rho_species, rhoMatrix,
   sphericity <- surfaceareaperfectsphere/surfaceareaparticle
   Psi <- sphericity/circularity # Shape factor Dioguardi
-  CSF <- Shortest_side/(sqrt(Longest_side*Intermediate_side)) #Corey Shape Factor
+  CSF <- rad_species/(sqrt(Longest_side*Intermediate_side)) #Corey Shape Factor
```
For Bagheri, more input is required (kS, kN) for f_DragCoefficient:
 f_DragCoefficient <- function(DragMethod, Re, Psi, CSF, kS, kN)
However, they are not defined here. If you allow me I can add the calculations for kS and kN here.

jormercury replied:
Ok, I think in the update vignette, we need to show tests with all the drag methods in order to catch such omissions! I can have a first try tomorrow. But help is welcome.

nadimsaadi commented on R/f_SurfaceAreaParticle.R:
```diff
@@ -1,50 +1,53 @@
+    a <- Longest_side / 2
+    b <- Intermediate_side / 2
+    c <- Shortest_side / 2
+    z <- 1.6075
```
<img width="425" height="620" alt="image" src="https://github.com/user-attachments/assets/23208213-45f1-4e55-a47f-9f8331b0d2cf" />

a, b and c are defined are Longest, Intermediate, shortest (without /2) in Valerie's report. I can adjust accordingly if you agree.

jormercury replied:
I changed the /2 for the '_sides'. But , please do check that the formula is correct (from report and in the code), eventually good to have a figure like this in the documentation I think.

nadimsaadi reviewed (commented):
I think there are inconsistencies overall in the perimeter, surface area and volume calculations. If you agree, I'm happy to go over all of them and double check.
Further, the parameters needed for Bagheri are not calculated in f_settlingvelocity (see comment).

jormercury committed:
Fix, twice devision by 2.

nadimsaadi committed:
Check of perimeter, surface area and volume calculations

I double checked the calculations for perimeter, surface area and volume, required for velocity calculation.

nadimsaadi committed:
Integration of Bagheri equation and parameters

Added in each file the computation of the parameters required for bagheri (kS and kN) and described them.
Made sure the function for drag and setvelsolver are called using kS and kN.

nadimsaadi commented:
I corrected the inconsistencies for parameter calculation, and made sure the implementation of Bagheri works.
However, I am not sure how to implement a code that calculates k using different methods and plots the results for comparison, but I do agree it would be useful to test. I can look into it if you don't have much time, but could you please point me a code that I could replicate?
Thanks.

jormercury committed:
Merge branch 'development' into Shape_setvel

jormercury commented:
> I corrected the inconsistencies for parameter calculation, and made sure the implementation of Bagheri works. However, I am not sure how to implement a code that calculates k using different methods and plots the results for comparison, but I do agree it would be useful to test. I can look into it if you don't have much time, but could you please point me a code that I could replicate? Thanks.

Great! Here is the script that compares the outcome for default implementation with this and the development branch: https://github.com/rivm-syso/SBooScripts/blob/b57bfad15e99539d97f4bcab58272b339608f50e/vignettes/Development/20250826_Plastics_Shape_update.Rmd

I think that we do not really need to compare with another version, as we verify already for spherical/default settling. But can just compare different runs of the 'new' version. So for the same particle with different shapes and methods. E.g. using MutateVars etc. and saving the k values and comparing those?

nadimsaadi committed:
Update f_SetVelWater.R

nadimsaadi committed:
Update f_SetVelWater.R

nadimsaadi committed:
Update .gitignore

nadimsaadi committed:
Update v_SettlingVelocity.R

nadimsaadi committed:
addition of "fiber" to rho and rad species calculation

nadimsaadi committed:
v_rad_species calls fVol with RadS

the fVol function was called within v_rad_species without specifying the radius. This would lead to an error, as no default radius is provided in the f_Vol.R function.

nadimsaadi committed:
fVol called with rad_particle specified

Same thing here, fVol was called without specifying the rad_particle which led to an error.

jormercury committed:
Change to using d_eq/2 instead of introducing rad_particle for consistency, rad_particle is already a name of such a variable in several functions.

jormercury committed:
Error checking and option to not set rad_particle

jormercury committed:
rad_particle not needed for 'new' shapes. and some checks implemented

jormercury committed:
layout

jormercury commented:
@nadimsaadi thanks for the additions and fixes. I also did some checks and small changes. There were still some issues with some of the shapes. All should be ok now.
